docs/design/fabric-v2+/trusted_ledger.md

```diff
@@ -0,0 +1,171 @@
+    - replay protection
+    - consistent view / snapshot during tx processing on integrity metadata / ledger state
+
+### Ledger maintainance
+
+Channel configuration:
+- Parse channel definition
+    - parse consensus definition
+- parse channel reader/writers
+- validate block signatures (Note that this check implies correctness as onfig blocks are validated by the ordering service ([see code](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/orderer/common/msgprocessor/standardchannel.go#L131)))
+- Maintain msp metadata for signature validation
+- parse Lifecycle policy 
+    - validate Lifecycle policy
+    We only support: a majority of Org.member (See [Link](https://hyperledger-fabric.readthedocs.io/en/release-2.2/chaincode_lifecycle.html#install-and-define-a-chaincode))
+
+Access control:
+- the "reader/writer/admin" access check is performed through endorsement, that is, if a valid endorsement for a transaction exists, this implies that the access check was successfully performed by the endorsers; the consortium (e.g., their admins) must ensure that the endorsement policy for a chaincode does not ruling out the authorization policy.
+- TLCC only verifies endorsement policies and thereby implicitly the "writers" check is performed by the endorsers
+
+MSP:
+- One Org - One MSP mapping - One root cert
+- Implement X509-based MSP
+    - Restrict:
+        - no intermediate
+        - no CRLs
+        - only support member role
+        - any certificate will match to role member
+
+Endorsing:
+- Phase1: Designated enclave only
+- Phase2: Any enclave that runs a particular FPC chaincode
+
+Versioning:
+- single autonomously monotonously increasing version number (TBD)
+
+Non-FPC Validation:
+- Transaction submitter identity validation
+    - submitter satisfies channel's writes policy
+- ERCC endorsement signatures verification
+- ERCC endorsement policy validation
+    - we only support (explicityly) Majority endorsement policy
+    - Restrict to ERCC namespace only
+
+FPC Validation:
+
+- Support for (subset of) endorsing policies
+    - FPC Chaincode (see above)
+- Parse chaincode definitions
+- Transaction submitter identity validation
+    - submitter satisfies channel's writes policy
+- FPC endorsement policy validation
+    - Phase 1: Support only OR with a single ORG.member
+    - Phase 2: Support only OR with multi ORG.member
+- FPC endorsement signatures
+
+
+## Development plan
+
+### short term
+See approach above
+
+We restrict the supported endorsement policies for lifecycle, ERCC, and FPC chaincodes.
+
+### mid/long term
+- Introduce FPC transaction type (similar to introduce FPC namespace) and create
+a dedicated FPC tx processor; (removes the need of custom validation plugins and interference with existing Fabric validation logic; and also gives more freedom to FPC validation logic as no it not longer bound to the structure and format of endorsement transaction).
+
```

Comment:
Related to above i guess is also to make tlcc maintain _all_ state, not only meta-data (thereby also removing any issue of synchronicity between peer and tlcc (at least as long as we do not go to cc2cc between fpc and non-fpc chaincode ...)